application_code/functions/java/function-astradb-writer/pom.xml

```diff
@@ -0,0 +1,202 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <groupId>com.example</groupId>
+        <artifactId>pulsarworkshop</artifactId>
+        <version>1.0.0</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>function-astradb-writer</artifactId>
+    <packaging>jar</packaging>
+
+    <properties>
+        <datastax.version>4.14.1</datastax.version>
+        <git-commit-id-plugin.version>4.9.10</git-commit-id-plugin.version>
+        <junit.version>4.11</junit.version>
+        <lombok-maven-plugin.version>1.18.20.0</lombok-maven-plugin.version>
+        <lombok.version>1.18.22</lombok.version>
+        <maven-assembly-plugin.version>3.1.1</maven-assembly-plugin.version>
+        <maven-compiler-plugin.version>3.10.1</maven-compiler-plugin.version>
+        <maven-model.version>3.3.9</maven-model.version>
+        <maven-remote-resources-plugin.version>3.0.0</maven-remote-resources-plugin.version>
+        <maven.compiler.release>17</maven.compiler.release>
+        <mockito-all.version>1.9.5</mockito-all.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <slf4j-nop.version>1.7.26</slf4j-nop.version>
+    </properties>
+    <name>${project.artifactId}</name>
+    <description>
+        Pulsar workshop demo application - Pulsar producer with native client API.
+    </description>
+
+    <build>
```

Comment:
This POM seems overly complex and include some unnecessary parts. Can you please simplify this a bit? Check this as an example: https://github.com/datastax/pulsar_workshop/blob/master/application_code/functions/java/add-metadata/pom.xml

For example,
1) "maven-remote-resources-plugin" is not needed because functions doesn't rely on the "common-resource" pacakge
2) the parent pom.xml uses maven shade plug in to create a uber jar, we don't need maven-assembly-plugin here
3) I'm not sure what is the purpose of including maven-compiler-plugin here

Reply:
Great points.